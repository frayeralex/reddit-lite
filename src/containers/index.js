export { default as App } from './App/App';
export { default as TopBar } from './TopBar/TopBar';
