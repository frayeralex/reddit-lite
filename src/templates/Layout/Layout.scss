.Layout {
  display: flex;
  flex-direction: column;
  min-height: 100vh;
  background: rgba(0, 0, 0, 0.1);

  .main-section {
    flex-grow: 1;
  }
}
