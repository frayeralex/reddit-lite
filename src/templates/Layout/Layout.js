import React from 'react';
import PropTypes from 'prop-types';
import { TopBar } from 'containers';
import './Layout.scss';

const Layout = ({ children, header }) => {
  return (
    <div className="Layout">
      {header || <TopBar />}
      <main className="main-section">{children}</main>
    </div>
  );
};

Layout.propTypes = {
  header: PropTypes.instanceOf(TopBar),
};

export default Layout;
