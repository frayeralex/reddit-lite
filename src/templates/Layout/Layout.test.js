import React from 'react';
import Layout from './Layout';
import { TopBar } from 'containers';

describe('<Layout/>', function() {
  it('should render without crashing', function() {
    const component = shallow(<Layout />);
    expect(component).toMatchSnapshot();
  });
  it('should contain TopBar', function() {
    const component = shallow(<Layout />);
    expect(component).toMatchSnapshot();
    component.setProps({ header: <TopBar title="title" /> });
    expect(component).toMatchSnapshot();
  });
});
