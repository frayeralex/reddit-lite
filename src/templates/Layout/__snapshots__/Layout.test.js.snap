// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Layout/> should contain TopBar 1`] = `
<div
  className="Layout"
>
  <Connect(TopBar) />
  <main
    className="main-section"
  />
</div>
`;

exports[`<Layout/> should contain TopBar 2`] = `
<div
  className="Layout"
>
  <Connect(TopBar)
    title="title"
  />
  <main
    className="main-section"
  />
</div>
`;

exports[`<Layout/> should render without crushing 1`] = `
<div
  className="Layout"
>
  <Connect(TopBar) />
  <main
    className="main-section"
  />
</div>
`;
